### PR TITLE
Simplify and deprecate `commonerrors.WriteErrors`

### LINKED ISSUE #2750: Support `ordered` `insert`s for SQLite
### What should be done?

* The handler should set `_id`s in documents and validate them. The backend should receive only valid documents.
* The handler should support `ordered` `insert`s.
* Tests should pass.

Batching will be implemented in #3271.

### Where?

See #3223.

### Definition of Done

- the handler updated;
- integration/compatibility tests un-skipped.

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Closes #2750.

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/commonerrors/write.go [hunk 1]
```diff
@@ -21,13 +21,28 @@ import (
 	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
+// writeError represents protocol write error.
+// It required to build the correct write error result.
+// The index field is optional and won't be used if it's nil.
```
Comment (chilagrow) [inline internal/handlers/commonerrors/write.go:26]: What does it mean by nil index field? It's int32 :thinking:

**Review by chilagrow — COMMENTED** [review]
There is a linter warning for deprecated function

File: internal/handlers/commonerrors/write.go (hunk 1 above)
Comment (AlekSi) [reply internal/handlers/commonerrors/write.go]: That was an old comment; updated

**Comment by AlekSi** [comment]
> There is a linter warning for deprecated function

Unfortunately, we can't fix it in this PR because of the way we use that linter and two separate configurations for golangci-lint. That code is considered new (it does not exist in the `main` branch), and our config disallows the usage of deprecated symbols in the new code.

That will not be a problem once this PR is merged.

**Review by chilagrow — APPROVED** [review]
Looks good to me :rocket:

**Pull request merged**